tests/milo/promotions.feature.test.js

```diff
@@ -359,4 +359,166 @@ test.describe('Promotions feature test suite', () => {
+      await expect(await PROMO.textDefault).toContainText(data.textDefault);
+    });
+
+    await test.step('Validate content insert before text', async () => {
+      await expect(await PROMO.textInsertBeforeCommon).toBeVisible();
+      await expect(await PROMO.textInsertBeforeCommon).toContainText(data.textBeforeText);
+    });
+
+    await test.step('Validate other promo filter actions are not applied', async () => {
+      await expect(await PROMO.textInsertAfterMarquee).not.toBeVisible();
+      await expect(await PROMO.marqueeReplace).not.toBeVisible();
+    });
+
+    await test.step('Go to the test page in DE locale', async () => {
+      testPage = `${baseURL}${data.CO_DE}${features[11].path}${miloLibs}`;
+      console.info('[Test Page][DE]: ', testPage);
+      await page.goto(testPage);
+      await page.waitForLoadState('domcontentloaded');
+    });
+
+    await test.step('Verify page filter on DE page', async () => {
+      await expect(await PROMO.marqueeDefault).toBeVisible();
+      await expect(await PROMO.textDefault).toBeVisible();
+      await expect(await PROMO.textInsertBeforeCommonDE).toBeVisible();
+      await expect(await PROMO.textInsertBeforeCommonDE).toContainText(data.textBeforeTextDE);
+      await expect(await PROMO.textInsertAfterMarquee).not.toBeVisible();
+      await expect(await PROMO.marqueeReplace).not.toBeVisible();
+    });
+
+    await test.step('Go to the test page in FR locale', async () => {
+      testPage = `${baseURL}${data.CO_FR}${features[11].path}${miloLibs}`;
+      console.info('[Test Page][FR]: ', testPage);
+      await page.goto(testPage);
+      await page.waitForLoadState('domcontentloaded');
+    });
+
+    await test.step('Verify page filter on FR page', async () => {
+      await expect(await PROMO.marqueeDefault).toBeVisible();
+      await expect(await PROMO.textDefault).toBeVisible();
+      await expect(await PROMO.textInsertBeforeCommonFR).toBeVisible();
+      await expect(await PROMO.textInsertBeforeCommonFR).toContainText(data.textBeforeTextFR);
+      await expect(await PROMO.textInsertAfterMarquee).not.toBeVisible();
+      await expect(await PROMO.marqueeReplace).not.toBeVisible();
+    });
+  });
+
+  // @Promo-remove-fragment - Validate fragment marquee remove
+  test(`${features[12].name},${features[12].tags}`, async ({ page, baseURL }) => {
+    const testPage = `${baseURL}${features[12].path}${miloLibs}`;
+    console.info('[Test Page]: ', testPage);
+
+    await test.step('Go to the test page', async () => {
+      await page.goto(testPage);
+      await page.waitForLoadState('domcontentloaded');
+    });
+
+    await test.step('Verify default test page content is not visible', async () => {
+      await expect(await PROMO.marqueeFragment).not.toBeVisible();
+    });
+  });
+
+  // @Promo-insert-fragment - Validate promo insert text after and before fragment
+  test(`${features[13].name},${features[13].tags}`, async ({ page, baseURL }) => {
```

Comment:
I think we can add the test name here instead of ${features[x].name}. It'll make it more clear when looking at the test file to see what the test case is verifying.
```suggestion
  test(`@Promo-insert-fragment }, ${features[13].tags}`, async ({ page, baseURL }) => {
```

Reply:
thanks @Dli3. Will update in my next PR all our tests, it is indeed more convenient.